### DISCUSSION TITLE
showBZDemo: This error is because RegionIntersection cannot work well for certain parameters in mathematica v11.2, but this problem does not occur in v12.0.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#33] [General]
I tried to run `showBZDemo["CubiBody"]`, but failed as follows:

![image](https://user-images.githubusercontent.com/11155854/152644895-11dbbe94-bb36-4550-840e-e8cbcc5ac26c.png)

![image](https://user-images.githubusercontent.com/11155854/152665199-eda8600e-3455-4a1d-ae02-14a014002b74.png)

As you can see, Mathematica says:

```
showBZDemo: This error is because RegionIntersection cannot work well for certain parameters in mathematica v11.2, but this problem does not occur in v12.0.
```
But the version I'm using is 12.3:

![image](https://user-images.githubusercontent.com/11155854/152644950-24ac4732-0601-4481-ad2d-6ef29641c9ff.png)

The above examples come from the usage shown [here](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/examples/document%20and%20examples.nb) and [here](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/examples/test.nb). Any hints for this problem will be highly appreciated.

Regards,
HZ

**Comment by hongyi-zhao** [comment]
Upgrading to the latest version solves this problem:

```mathematica
Clear["*`"]
<< SpaceGroupIrep`
BCHighSymKpt // Keys // InputForm
showBZDemo["CubiBody"]
showBZDemo["TetrBody(b)"]
```
![image](https://user-images.githubusercontent.com/11155854/152666614-1e644022-c880-4460-a91a-0cc572ffa5db.png)

![image](https://user-images.githubusercontent.com/11155854/152666713-c4e815e4-4195-44cf-8417-a80d6f4293e0.png)


![image](https://user-images.githubusercontent.com/11155854/152666590-14558371-2b78-45b0-932c-6b5b1ccf0064.png)

**Comment by goodluck1982** [comment]
In fact, `showBZDemo` works fine under version 11.2 with default basic vectors. e.g.
```mathematica
showBZDemo["OrthBody(c)"]
```

The `RegionIntersection` related problem only occurs when user-specified basic vectors are used, such as
```mathematica
showBZDemo["OrthBody(c)",  BasicVectors["OrthBody"] /. {a -> 2, b -> 3.2, c -> 5}]
```

It's strange that the error occurs without use-specified basic vectors, and it's especially strange that it works in v12.0 but not in v12.3. Confronting such bug originated from the underlying code of the language itself, I feel very helpless.

**Comment by hongyi-zhao** [comment]
> Confronting such bug originated from the underlying code of the language itself, I feel very helpless.

This is why open-source based implementations of scientific software projects are so popular in academia and relatively easy to solve low-level code problems.
